src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java

```diff
@@ -351,5 +352,76 @@ static UMessageType getMessageType(String ce_type){
                 return UMessageType.UMESSAGE_TYPE_UNSPECIFIED;
         }
     }
+    /**
+     * Get the UMessage from the cloud event
+     * @param event The CloudEvent containing the data.
+     * @return returns the UMessage
+     */
+    static UMessage toMessage(CloudEvent event) {
+        UUri source = LongUriSerializer.instance().deserialize(getSource(event));
+        UPayload payload = UPayload.newBuilder().setFormat(UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF)
```

Comment:
Format is only protobuf if datacontenttype is missing, otherwise you need to map it before you copy it.

Reply:
Done